src/CacheManager.Core/Internal/BaseCache.cs

```diff
@@ -407,6 +410,55 @@ public virtual bool Remove(string key, string region)
         }
 
         /// <summary>
+        /// Gets all keys that match the given pattern.
+        /// </summary>
+        /// <param name="pattern">A glob to match against keys.</param>
+        /// <param name="region">The cache region.</param>
+        /// <returns>The matching unique keys in the cache</returns>
+        public virtual IEnumerable<string> Keys(string pattern, string region)
+        {
+            var keys = AllKeys();
```

Comment:
Yeah, that might be fine. 
But if you return an IEnumerable, you should also implement it properly and not iterate over all keys everytime. IEnumerable allows the consumer to stop consumption at any point (e.g. .First() should iterate only until it finds the first match). That's why you should implement that with yield return everywhere and without copying the result into a list.